src/decorator.ts

```diff
@@ -55,10 +55,14 @@ export abstract class Decorator extends PathTransformVisitor {
     return not(isLibrary);
   }
 
-  abstract get name(): string;
+  get decoratorMatcher(): (node: DecoratorNode) => boolean {
+    return (node: DecoratorNode) => decorates(node, this.name)
+  }
```

Comment:
why not 
```ts
decoratorMatcher(node: DecoratorNode) => boolean {
    return  decorates(node, this.name)
  }
```

Reply:
And `DecoratorVisitor` is not implemented.

Reply:
Because of the binding of `this`.  So if you want to use this function in a filter it wouldn't work in yours, but the property returns a new function that has `this` bound to it. e.g. `decorators.filter(visitor.decoratorMater)`.

And yeah, oops on the lack of implementation.  I can't recall why I added that interface.